index.js

```diff
@@ -38,37 +50,19 @@ function setIgnore(opts, negatives, positiveIndex) {
 	return opts;
 }
 
-module.exports = function (patterns, opts, cb) {
+module.exports = function (patterns, opts) {
 	var sortedPatterns = sortPatterns(patterns);
-
-	if (typeof opts === 'function') {
-		cb = opts;
-		opts = {};
-	}
+	opts = opts || {};
 
 	if (sortedPatterns.positives.length === 0) {
 		cb(null, []);
 		return;
 	}
 
-	async.parallel(sortedPatterns.positives.map(function (positive) {
-		return function (cb2) {
-			glob(positive.pattern, setIgnore(opts, sortedPatterns.negatives, positive.index), function (err, paths) {
-				if (err) {
-					cb2(err);
-					return;
-				}
-
-				cb2(null, paths);
-			});
-		};
-	}), function (err, paths) {
-		if (err) {
-			cb(err);
-			return;
-		}
-
-		cb(null, arrayUnion.apply(null, paths));
+	return Promise.all(sortedPatterns.positives.map(function (positive) {
+		return globP(positive.pattern, setIgnore(opts, sortedPatterns.negatives, positive.index));
+	})).then(function (paths) {
+		return arrayUnion.apply(null, paths);
```

Comment:
Are you sure? Without the return the promise won't yield the list of deleted paths.
```
   1) should glob - async
   2) should glob with multiple patterns - async
   3) should respect patterns order - async
   ✓ should glob - sync
   ✓ cwd option
   ✓ should not mutate the options object - async
   ✓ should not mutate the options object - sync

  4 passing (29ms)
  3 failing

  1)  should glob - async:
     AssertionError: undefined deepEqual [ 'a.tmp', 'b.tmp', 'c.tmp', 'd.tmp', 'e.tmp' ]
      at test.js:27:10

  2)  should glob with multiple patterns - async:
     AssertionError: undefined deepEqual [ 'a.tmp', 'b.tmp' ]
      at test.js:33:10

  3)  should respect patterns order - async:
     AssertionError: undefined deepEqual [ 'a.tmp' ]
      at test.js:39:10
```